In [14]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [8]:
df = pd.read_csv('emotions_train.csv')
df.shape

(21459, 2)

In [12]:
df.columns

Index(['Text', 'Emotion'], dtype='object')

Le jeu de données fournis est composé de deux colonnes : Text qui rassemble les postes des utilisateurs d'un réseau social et Emotion qui associe les postes à une émotion.
Nous disposons de 21459 enregistrements.

In [5]:
df.info

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21459 entries, 0 to 21458
Data columns (total 2 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   Text     21459 non-null  object
 1   Emotion  21459 non-null  object
dtypes: object(2)
memory usage: 335.4+ KB


Il n'y a aucune donnée manquante

In [13]:
df['Emotion'].unique()

array(['sadness', 'anger', 'love', 'surprise', 'fear', 'joy'],
      dtype=object)

Les différentes émotions répertoriées sont : la tristesse, la colère, l'amour, la surprise, la peur et la joie.

In [27]:
df['Emotion'].value_counts()

Emotion
joy         7029
sadness     6265
anger       2993
fear        2652
love        1641
surprise     879
Name: count, dtype: int64